daemon/src/main/java/org/lsposed/lspd/cli/Utils.java

```diff
@@ -0,0 +1,69 @@
+package org.lsposed.lspd.cli;
+
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.RemoteException;
+
+import com.beust.jcommander.Parameters;
+
+import org.lsposed.lspd.ILSPManagerService;
+import org.lsposed.lspd.models.Application;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+
+public class Utils {
+
+    public static final String CMDNAME = "cli";
+
+    public enum ERRCODES {
+        NOERROR,
+        USAGE,
+        EMPTY_SCOPE,
+        ENABLE_DISABLE,
+        SET_SCOPE,
+        NO_DAEMON,
+        REMOTE_ERROR,
+        NO_XPOSEDMODULE,
+        NO_PACKAGE
+    }
+
+    private static HashMap<String,PackageInfo> packagesMap;
+
+    private static void initPackagesMap(ILSPManagerService managerService) throws RemoteException {
+        var packages =
+                managerService.getInstalledPackagesFromAllUsers(PackageManager.GET_META_DATA | PackageManager.MATCH_UNINSTALLED_PACKAGES, true).getList();
+        packagesMap = new HashMap<>();
+        for (var packageInfo: packages) {
+            packagesMap.put(packageInfo.packageName, packageInfo);
+        }
+    }
+
+    public static boolean validXposedModule(ILSPManagerService managerService, String packageName) throws RemoteException {
```

Comment:
I think these could be done by the daemon. The daemon can just report an error and send it to cli.

Reply:
with last commit should be ok